bitnami/common/templates/validations/_mongodb.tpl

```diff
@@ -0,0 +1,84 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Validate MongoDB required passwords are not empty.
+
+Usage:
+{{ include "common.validations.values.mongodb.passwords" (dict "secret" "secretName" "subchart" false "context" $) }}
+Params:
+  - secret - String - Required. Name of the secret where MongoDB values are stored, e.g: "mongodb-passwords-secret"
+  - subchart - Boolean - Optional. Whether MongoDB is used as subchart or not. Default: false
+*/}}
+{{- define "common.validations.values.mongodb.passwords" -}}
+  {{- $existingSecret := include "common.mongodb.values.auth.existingSecret" . -}}
+  {{- $enabled := include "common.mongodb.values.enabled" . -}}
+  {{- $authPrefix := include "common.mongodb.values.key.auth" . -}}
+  {{- $valueKeyRootPassword := printf "%s.rootPassword" $authPrefix -}}
+  {{- $valueKeyUsername := printf "%s.username" $authPrefix -}}
+  {{- $valueKeyPassword := printf "%s.password" $authPrefix -}}
+  {{- $valueKeyReplicaSetKey := printf "%s.replicaSetKey" $authPrefix -}}
+
+  {{- if and (not $existingSecret) (eq $enabled "true") -}}
```

Comment:
MongoDB has an additional parameter named `auth.enabled` in order to check if the general authentication is enabled. Else, no passwords are generated/required.

See
https://github.com/bitnami/charts/blob/f5abfb8e423872199518733bde449d57392eab0a/bitnami/mongodb/templates/_helpers.tpl#L137-L139

https://github.com/bitnami/charts/blob/f5abfb8e423872199518733bde449d57392eab0a/bitnami/mongodb/templates/secrets.yaml#L1-L3

So here we should be checking:

There is no existingSecret, the mongoDB itself is enabled and the authentication for mongoDB is enabled
```suggestion
  {{- if and (not $existingSecret) (eq $enabled "true") (eq $authEnabled "true") -}}
```